lib/ts/recipe/session/sessionFunctions.ts

```diff
@@ -28,12 +28,30 @@ import { logDebugMessage } from "../../logger";
 export async function createNewSession(
     helpers: Helpers,
     userId: string,
+    disableAntiCsrf: boolean,
     accessTokenPayload: any = {},
     sessionData: any = {}
 ): Promise<CreateOrRefreshAPIResponse> {
     accessTokenPayload = accessTokenPayload === null || accessTokenPayload === undefined ? {} : accessTokenPayload;
     sessionData = sessionData === null || sessionData === undefined ? {} : sessionData;
 
+    if (
+        !disableAntiCsrf &&
+        helpers.config.cookieSameSite === "none" &&
+        !helpers.config.cookieSecure &&
+        !(
+            (helpers.appInfo.topLevelAPIDomain === "localhost" || isAnIpAddress(helpers.appInfo.topLevelAPIDomain)) &&
+            (helpers.appInfo.topLevelWebsiteDomain === "localhost" ||
+                isAnIpAddress(helpers.appInfo.topLevelWebsiteDomain))
+        )
+    ) {
+        // We can allow insecure cookie when both website & API domain are localhost or an IP
+        // When either of them is a different domain, API domain needs to have https and a secure cookie to work
+        throw new Error(
+            "Since your API and website domain are different, for sessions to work, please use https on your apiDomain and dont set cookieSecure to false."
+        );
+    }
+
```

Comment:
the placement of this is strange. I think it should be in the recipeImplementation's createNewSession, and it should be called if transfer method is cookie. Right now, you are determining if this check should run based on the value of  `disableAntiCsrf` which is very indirect.